### ISSUE TITLE
Question about lingvo's language model and LM fusion in ASR: how to use changed LM to finish ASR LM fusion task

### ISSUE THREAD

**Issue by iamxiaoyubei** [tensorflow/lingvo#103]
Hi, recently, after I edited the ComputeLogitsWithLM and FProp function for LM fusion(in beam search while inference), I plan to **train a language model to use in LM fusion**. 

（ I use Librispeech960Wpm model in ASR task)

**I hope my language model can calculate the maximum possible logit based on the previous logit.**

Just complete the task of the **LM** function:
```
logits = logits + ratio * LM(prev_logits)
```

I tried to finish my task with some existing language model. However, I found that these existing ones could not be used. It is because logit is the probability of word piece which dimension is 16328. So I should predict the next **word piece** based on the current predicted **word piece**(not word).

Therefore, I modify the **WordLevelOneBwdsSimpleSampledSoftmax** model in lingvo to use **WpmTokenizer and train on librispeech text corpus**. But I am confused about the [fetch values](https://github.com/tensorflow/lingvo/blob/c8b6df48df825bbb51e8f59e0565228432e6fa82/lingvo/tasks/lm/model.py#L201). I can't be sure **what they are and how to use this fetch values on my LM fusion task.**

So, could you give me some advice on **how to use your Language model in ASR LM fusion task?**(for example, how to use your LM to predict next logits based on prev logits?)

Besides, if I have language model to predict next wordpiece, **how can I call this model in the code to get the next wordpiece probability while ASR inference**?

It seems that it is not possible to import graph, load ckpt and run language model directly in the **ComputeLogitsWithLM** function. And, in the **inference code**:
```
import tensorflow as tf
tf.app.flags.DEFINE_string('f', '', 'kernel')
with open('/tmp/librispeech/single_sample/61-70968-0000.wav') as f:
    read_data = f.read()

# import datetime
import time;
from lingvo import model_imports
from lingvo import model_registry
from lingvo.core import inference_graph_exporter
from lingvo.core import predictor
from lingvo.core.ops.hyps_pb2 import Hypothesis
checkpoint = tf.train.latest_checkpoint('/tmp/librispeech/log/train')
print('Using checkpoint %s' % checkpoint)

# Run inference
params = model_registry.GetParams('asr.librispeech.Librispeech960Wpm', 'Test')
inference_graph = inference_graph_exporter.InferenceGraphExporter.Export(params)
pred = predictor.Predictor(inference_graph, checkpoint=checkpoint)
convert_start = time.time()
# hyps, src_frames, en_frames, scores = pred.Run(['hypotheses', 'src_frames', 'encoder_frames', 'scores'], wav=read_data)
hyps, scores = pred.Run(['hypotheses', 'scores'], wav=read_data)
convert_end = time.time()
diff = convert_end - convert_start
print("Done predict:{}s".format(diff))
print(hyps)
# print(src_frames)
# print(en_frames)
print(scores)
```
It seems that there is also no way to directly import model, load ckpt and then passed language model into the ASR model.

Look forward to your reply. Thank you!
@jonathanasdf @rprabhavalkar @ruomingp @bignamehyp

**Comment by iamxiaoyubei** [comment]
Besides, if I trained a language model with lingvo, how can I call this model in the code to get the next wordpiece probability while ASR inference?

It seems that it is not possible to import graph, load ckpt and run model directly in the **ComputeLogitsWithLM** function. But, in the inference code:
```
import tensorflow as tf
tf.app.flags.DEFINE_string('f', '', 'kernel')
with open('/tmp/librispeech/single_sample/61-70968-0000.wav') as f:
    read_data = f.read()

# import datetime
import time;
from lingvo import model_imports
from lingvo import model_registry
from lingvo.core import inference_graph_exporter
from lingvo.core import predictor
from lingvo.core.ops.hyps_pb2 import Hypothesis
checkpoint = tf.train.latest_checkpoint('/tmp/librispeech/log/train')
print('Using checkpoint %s' % checkpoint)

# Run inference
params = model_registry.GetParams('asr.librispeech.Librispeech960Wpm', 'Test')
inference_graph = inference_graph_exporter.InferenceGraphExporter.Export(params)
pred = predictor.Predictor(inference_graph, checkpoint=checkpoint)
convert_start = time.time()
# hyps, src_frames, en_frames, scores = pred.Run(['hypotheses', 'src_frames', 'encoder_frames', 'scores'], wav=read_data)
hyps, scores = pred.Run(['hypotheses', 'scores'], wav=read_data)
convert_end = time.time()
diff = convert_end - convert_start
print("Done predict:{}s".format(diff))
print(hyps)
# print(src_frames)
# print(en_frames)
print(scores)
```
It seems that there is also no way to directly import model, load ckpt and then passed into the ASR model.
